storage/mutable_segment.go

```diff
@@ -100,7 +101,8 @@ func newMutableSegment(
 		SetIntArrayPool(opts.IntArrayPool()).
 		SetDoubleArrayPool(opts.DoubleArrayPool()).
 		SetStringArrayPool(opts.StringArrayPool()).
-		SetInt64ArrayPool(opts.Int64ArrayPool())
+		SetInt64ArrayPool(opts.Int64ArrayPool()).
+		SetStringValuesResetFn(strings.ValuesResetFn)
```

Comment:
`ResetArrayFn`.

Reply:
Also add a comment here to say this is to avoid holding on to documents received after the array is returned to pool.